content/en/docs/topics/kubernetes_apis.md

```diff
@@ -0,0 +1,139 @@
+  will be removed in Kubernetes 1.16):
+    - Follow the step 1 procedure
+  - Otherwise (for example, you are already running a Kubernetes version where
+  some API versions reported by `helm get manifest` are no longer available):
+    - You need to edit the release manifest in-cluster updating the API
+    versions to supported APIs. See
+    [how tell Helm that the release now uses a new API version](#updating-api-versions-of-a-release-manifest)
+    for more details
+
+> Note: In all cases of updating a Helm release with supported APIs, you should
+never rollback the release to a version prior to the release version with the
+supported APIs.
+
+> Recommendation: The best practice is to upgrade releases using deprecated API
+versions to supported API versions, prior to upgrading to a kubernetes cluster
+that removes those API versions. 
+
+If you don't update a release as suggested previously, you will have an error
+similar to the following when trying to upgrade a release in a Kubernetes version
+where its API version(s) is/are removed:
+
+```
+Error: UPGRADE FAILED: current release manifest contains removed kubernetes api(s)
+for this kubernetes version and it is therefore unable to build the kubernetes
+objects for performing the diff. error from kubernetes: unable to recognize "":
+no matches for kind "Deployment" in version "apps/v1beta1"
+```
+
+Helm fails in this scenario because it attempts to create a diff patch between
+the current deployed release (which contains the Kubernetes APIs that are removed
+in this Kubernetes version) against the chart you are passing with the
+updated/supported API versions. The underlying reason for failure is that when
+Kubernetes removes an API version, the Kubernetes Go client library can no longer
+parse the deprecated objects and Helm therefore fails when calling the library.
+Helm unfortunately is unable to recover from this situation and is no longer able
+to manage such a release.
+See [how tell Helm that the release now uses a new API version](#updating-api-versions-of-a-release-manifest)
+for more details on how to recover from this scenario.
+
+## Updating API Versions of a Release Manifest
+
+The manifest is a property of the Helm release object which is stored in the data
+field of a Secret (default) or ConfigMap in the cluster. The data field contains
+a gzipped object which is base 64 encoded twice. There is a Secret/ConfigMap per
+release version/revision in the namespace of the release.
+
+You can use the Helm [mapkubeapis](https://github.com/hickeyma/helm-mapkubeapis)
+plugin to perform the update of a release to supported APIs. Check out the
+readme for more details.
+
+Alternatively, you can follow these manual steps to perform an update of the API
+versions of a release manifest:
+
+- Get latest deployed release:
+`kubectl get secret <release_version_secret_name> -n <release_version_namespace> -o yaml > release.bak`
```

Comment:
Do you think we should add a step to say how to find the secret name?

Reply:
ok, good idea. Added.